Enable poro solid interaction in new time integration

SchaefP1 opened the pull request:
<!--
* Title: Provide a general summary of your changes in the Title above.

* Assignees:  If you know anyone who should likely handle bringing this pull request (PR) to completion -- that's usually yourself -- select them from the Assignees drop-down on the right.

* Labels: Update the label of the issue(s) addressed by this pull request to "Under Review".
-->

## Description and Context
Poro structure interaction (using poro and normal structure elements in the same simulation) is currently only possible with the old time integration. With the new time integration it leads to a segmentation fault as the discretization is set up new but this info is not passed on to the structure adapter. This pull request fixes this by using structure_setup() after setting up the discretization new. Additionally, some lines where removed from the restart function of poro that else caused issues and seem unneeded (duplicated and do not change the tests).

The tests are identical to the ones from the old time integration only using the new time integration (1 result is updated).

## Related Issues and Merge Requests
Part of #146

SchaefP1 commented on src/particle_rigidbody/4C_particle_rigidbody_result_test.cpp:
```diff
@@ -12,6 +12,8 @@
 #include "4C_particle_rigidbody_interface.hpp"
 #include "4C_utils_exceptions.hpp"
 
+#include <algorithm>
```
This change is unrelated to the rest of the pull request but without this include main did not compile for me.

sebproell replied:
Out of curiosity: which compiler are you using?

SchaefP1 replied:
gcc-14.2.0

sebproell reviewed (approved):
The C++ side is fine, although I have to admit I have a hard time understanding how the change enables a new feature.

amgebauer commented on tests/input_files/poro_3D_psi_hex8_new_struc.dat:
```diff
@@ -0,0 +1,961 @@
+NODE 215 COORD 5.0000000000000000e+00 5.0000000000000000e+00 -3.0000000000000000e+00
+NODE 216 COORD 5.0000000000000000e+00 5.0000000000000000e+00 -5.0000000000000000e+00
+------------------------------------------------STRUCTURE ELEMENTS
+1 SOLIDH8PORO HEX8 1 2 3 4 5 6 7 8 MAT 1 KINEM nonlinear EAS none
```
What kind of poro formulation is used in this testcase? Can we use the new solid poro elements?

SchaefP1 replied:
Pressure velocity based. The new solid poro elements also works and is tested in the other 3D test. I chose to include the old elements as well since this test exists in the old time integration.

lkoeglmeier replied:
I think the new solid poro elements also work for the old time integration, or am I understanding this wrong?

SchaefP1 replied:
Yes, both versions of the solid poro element work in the old time integration. The new test cases are just copies of the test cases in the old time integration with the main change being the use of the new time integration. The changes are independent of the used version of the solid element and I only included tests for both since both are tested in the old time integration. From my side we can also remove this test case if the migration to the new solid poro elements makes adding new tests with the old solid poro elements undesirable.

lkoeglmeier replied:
Ah sorry, now I get it. My fault, thanks for the explanation. I think we want to get rid of the old solid elements at some point, so from my point of view, it would be sufficient to have only the test cases with the new solid elements. But I am also fine if you or @amgebauer have a different opinion on this.

SchaefP1 replied:
I removed the test with the old solid elements.

amgebauer commented on tests/input_files/poro_3D_psi_hex8_new_struc.dat:
```diff
@@ -0,0 +1,961 @@
+TOLRES_PORO                     1e-04
+VECTORNORM_RESF                 L1
+VECTORNORM_INC                  Rms
+TIMESTEP                        0.01
+RESULTSEVRY                           1
```
This file itself is not super small and we are doing 100 computation steps, and writing output on every step.

Is this really necessary or might it be sufficient to

 * use a smaller test
 * larger timestep size / fewer timesteps
 * write output only on few points in time?

I haven't checked how long the test is running in the pipeline, but to me it looks like a little too complex and runtime could be reduced a lot with just a few modifications.

The same applies to the other 3D input file. Probably somehow related to #118.

mayrmt replied:
If not testing for output, there's for sure no need to write results in every time step.

amgebauer commented on tests/input_files/new_solidporo_pressure_velocity_based_ele_3D_psi_hex8_new_struc.dat:
```diff
@@ -0,0 +1,963 @@
+E 4 - NUMDOF 3 ONOFF 1 1 1 VAL 1.0 0.0 0.0 FUNCT 3 0 0
+// Surface x+
+E 6 - NUMDOF 3 ONOFF 1 1 1 VAL 0.0 0.0 0.0 FUNCT 0 0 0
+// Surface y+
+//E 5 - NUMDOF 3 ONOFF 1 1 1 VAL 0.0 0.0 0.0 FUNCT 0 0 0
+// Surface y-
+//E 3 - NUMDOF 3 ONOFF 1 1 1 VAL 0.0 0.0 0.0 FUNCT 0 0 0
+// Surface z+
+//E 1 - NUMDOF 3 ONOFF 1 1 1 VAL 0.0 0.0 0.0 FUNCT 0 0 0
+// Surface z-
+//E 2 - NUMDOF 3 ONOFF 1 1 1 VAL 0.0 0.0 0.0 FUNCT 0 0 0
```
```suggestion
```

amgebauer commented on tests/input_files/poro_3D_psi_hex8_new_struc.dat:
```diff
@@ -0,0 +1,961 @@
+E 4 - NUMDOF 3 ONOFF 1 1 1 VAL 1.0 0.0 0.0 FUNCT 3 0 0
+// Surface x+
+E 6 - NUMDOF 3 ONOFF 1 1 1 VAL 0.0 0.0 0.0 FUNCT 0 0 0
+// Surface y+
+//E 5 - NUMDOF 3 ONOFF 1 1 1 VAL 0.0 0.0 0.0 FUNCT 0 0 0
+// Surface y-
+//E 3 - NUMDOF 3 ONOFF 1 1 1 VAL 0.0 0.0 0.0 FUNCT 0 0 0
+// Surface z+
+//E 1 - NUMDOF 3 ONOFF 1 1 1 VAL 0.0 0.0 0.0 FUNCT 0 0 0
+// Surface z-
+//E 2 - NUMDOF 3 ONOFF 1 1 1 VAL 0.0 0.0 0.0 FUNCT 0 0 0
```
```suggestion
```

SchaefP1 commented:
> The C++ side is fine, although I have to admit I have a hard time understanding how the change enables a new feature.

@sebproell Thanks for the review! Basically, poroelastic PSI already works in the new time integration once the set-up is complete. The problems are only while setting it up and restarting it as some info is not correctly passed on to the new time integration and it calls evaluate during set-up while some poro specific things are not set yet. I added some comments to make it easier to understand why this is needed and also changed a bit how I handle this in the read_restart function.
